kernel/arch/dreamcast/include/dc/sq.h

```diff
@@ -22,73 +36,178 @@
-    \param  c               The value to set (in the low 8-bits).
+    \warning
+    The dest pointer must be a 32-byte aligned with n being a multiple of 32, 
+    and only the low 8-bits are used from c.
+
+    \param  dest            The address to begin setting at (32-byte aligned).
+    \param  src             The value to set (in the low 8-bits).
     \param  n               The number of bytes to set (multiple of 32).
     \return                 The original value of dest.
+
+    \sa sq_set16(), sq_set32(), sq_set_pvr()
 */
-void * sq_set(void *s, uint32 c, int n);
+void * sq_set(void *dest, uint32 c, int n);
 
-/** \brief  Set a block of memory to a 16-bit value.
+/** \brief   Set a block of memory to a 16-bit value.
+    \ingroup store_queues
 
     This function is similar to calling memset2(), but uses the store queues to
     do its work.
 
-    \param  s               The address to begin setting at (32-byte aligned).
+    \warning
+    The dest pointer must be a 32-byte aligned with n being a multiple of 32, 
+    and only the low 16-bits are used from c.
+
+    \param  dest            The address to begin setting at (32-byte aligned).
     \param  c               The value to set (in the low 16-bits).
     \param  n               The number of bytes to set (multiple of 32).
     \return                 The original value of dest.
+
+    \sa sq_set(), sq_set32(), sq_set_pvr()
 */
-void * sq_set16(void *s, uint32 c, int n);
+void * sq_set16(void *dest, uint32 c, int n);
 
-/** \brief  Set a block of memory to a 32-bit value.
+/** \brief   Set a block of memory to a 32-bit value.
+    \ingroup store_queues
 
     This function is similar to calling memset4(), but uses the store queues to
     do its work.
 
-    \param  s               The address to begin setting at (32-byte aligned).
+    \warning
+    The dest pointer must be a 32-byte aligned with n being a multiple of 32!
+
+    \param  dest            The address to begin setting at (32-byte aligned).
     \param  c               The value to set (all 32-bits).
     \param  n               The number of bytes to set (multiple of 32).
     \return                 The original value of dest.
+
+    \sa sq_set(), sq_set16(), sq_set_pvr()
+*/
+void * sq_set32(void *dest, uint32 c, int n);
+
+/** \brief   Clear a block of memory.
+    \ingroup store_queues
+
+    This function is similar to calling memset() with a value to set of 0, but
+    uses the store queues to do its work.
+
+    \warning
+    The dest pointer must be a 32-byte aligned with n being a multiple of 32!
+
+    \param  dest            The address to begin clearing at (32-byte aligned).
+    \param  n               The number of bytes to clear (multiple of 32).
+*/
+void sq_clr(void *dest, int n);
+
+/** \brief   PVR register governing texture memory write data size
+    \ingroup store_queues
+*/
+#define SQ_PVR_LMMODE (*(volatile uint32_t *)(void *)0xa05f6884)
```

Comment:
This probably has no good reason to be exposed publicly, especially not here (since it is not a store queue related register). Please move it to the .c file.

Reply:
Done